pkg/venafi/tpp/connector.go

```diff
@@ -1001,6 +1043,125 @@ func (c *Connector) SetPolicy(name string, ps *policy.PolicySpecification) (stri
 	return status, nil
 }
 
+func (c *Connector) setContact(tppPolicy *policy.TppPolicy, policyExists bool) (status string, err error) {
+
+	var contacts []string
+
+	if tppPolicy.Contact != nil {
+		contacts, err = c.resolveContacts(tppPolicy.Contact)
+		if err != nil {
+			return "", err
+		}
+	} else {
+		if !policyExists {
+			var currentIdentity *policy.IdentityEntry
+
+			currentIdentity, err = c.getCurrentIdentity()
+
+			if err != nil {
+				return "", err
+			}
+
+			if currentIdentity != nil {
+				contacts = append(contacts, currentIdentity.PrefixedUniversal)
+			} else {
+				return "", fmt.Errorf("the identity for the current user was not found")
+			}
+		}
+	}
+
+	if contacts != nil {
+		tppPolicy.Contact = contacts
+
+		_, status, _, err = createPolicyAttribute(c, policy.TppContact, tppPolicy.Contact, *(tppPolicy.Name), true)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	return status, nil
+}
+
+func (c *Connector) getCurrentIdentity() (*policy.IdentityEntry, error) {
+	var currentIdentity *policy.IdentityEntry
+
+	resp, err := c.identitySelf()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, identityEntry := range resp.Identities {
+		if identityEntry.Type == policy.IdentityUser {
+			currentIdentity = &identityEntry
+			break
+		}
+	}
+
+	return currentIdentity, nil
+}
+
+func (c *Connector) resolveContacts(contacts []string) ([]string, error) {
+	var identities []string
+	for _, contact := range contacts {
+		identity, err := c.getIdentity(contact)
+		if err != nil {
+			return nil, err
+		}
+		identities = append(identities, identity.PrefixedUniversal)
+	}
+
+	return identities, nil
+}
+
+func (c *Connector) getIdentity(userName string) (*policy.IdentityEntry, error) {
+	if userName == "" {
+		return nil, fmt.Errorf("identity string cannot be null")
+	}
+
+	req := policy.BrowseIdentitiesRequest{
+		Filter:       userName,
+		Limit:        2,
+		IdentityType: policy.AllIdentities,
+	}
+
+	resp, err := c.browseIdentities(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(resp.Identities) > 1 {
+		return nil, fmt.Errorf("only one Identity must be returned but it was returned more than one")
+	}
+
+	return &resp.Identities[0], nil
+}
+
+func (c *Connector) browseIdentities(browseReq policy.BrowseIdentitiesRequest) (*policy.BrowseIdentitiesResponse, error) {
+
+	statusCode, status, body, err := c.request("POST", urlResourceBrowseIdentities, browseReq)
+	if err != nil {
+		return nil, err
+	}
+	browseIdentitiesResponse, err := parseBrowseIdentitiesResult(statusCode, status, body)
+	if err != nil {
+		return nil, err
+	}
+	return &browseIdentitiesResponse, nil
+}
+
+func (c *Connector) identitySelf() (*policy.IdentitySelfResponse, error) {
```

Comment:
Maybe this function should be called selfIdentity instead of identitySelf, makes more sense

Reply:
got it... doing the change